### PR TITLE
Api doc update

### PR DESCRIPTION
**Pull request by MHBauer**
This is some small updates that are complete as it is. I can turn it into a long running WIP, but I'd rather we focus on small incremental additions for now.

@pmorie @deads2k @lavalamp Is `apis/authentication` still a good base to template off of for newcomers?

### CODE REVIEW

File: contributors/devel/adding-an-APIGroup.md [hunk 1]
```diff
@@ -21,8 +21,11 @@ in your group;
 
 2. Create pkg/apis/`<group>`/{register.go, `<version>`/register.go} to register
 this group's API objects to the encoding/decoding scheme (e.g.,
-[pkg/apis/authentication/register.go](../../pkg/apis/authentication/register.go) and
-[pkg/apis/authentication/v1beta1/register.go](../../pkg/apis/authentication/v1beta1/register.go);
+[pkg/apis/authentication/register.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/authentication/register.go) and
+[pkg/apis/authentication/v1beta1/register.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/authentication/v1beta1/register.go);
+The register files must have a var called SchemeBuilder for the
+generated code to reference. There must be an AddToScheme method for
+the installer to reference.
```
Comment (lavalamp) [inline contributors/devel/adding-an-APIGroup.md]: Optional: suggest you add something like, "You can look at a group under `pkg/apis/...` for example register.go files to use as a template, but do not copy the register.go files under `pkg/api/...`--they are not general."

**Comment by lavalamp** [comment]
consider this an lgtm from me with or without my suggested change

File: contributors/devel/adding-an-APIGroup.md (hunk 1 above)
Comment (MHBauer) [inline contributors/devel/adding-an-APIGroup.md]: agree based on my experiences. Do we have a blurb somewhere about the history of the api group definitions? It is my understanding that `api/` is the old stuff and `apis/` is the new stuff, but I don't have more specific information or understanding.

**Comment by liggitt** [comment]
> @pmorie @deads2k @lavalamp Is apis/authentication still a good base to template off of for newcomers?

I think so, it's reasonably small, and adheres to the k8s.io API group suffix

File: contributors/devel/adding-an-APIGroup.md [hunk 2]
```diff
@@ -21,21 +21,38 @@ in your group;
 
 2. Create pkg/apis/`<group>`/{register.go, `<version>`/register.go} to register
 this group's API objects to the encoding/decoding scheme (e.g.,
-[pkg/apis/authentication/register.go](../../pkg/apis/authentication/register.go) and
-[pkg/apis/authentication/v1beta1/register.go](../../pkg/apis/authentication/v1beta1/register.go);
-
-3. Add a pkg/apis/`<group>`/install/install.go, which is responsible for adding
-the group to the `latest` package, so that other packages can access the group's
```
Comment (MHBauer) [inline contributors/devel/adding-an-APIGroup.md]: Having written an api group recently, I have no idea what this `latest` and `latest.Group` text is talking about.

**Comment by MHBauer** [comment]
@lavalamp your text included. Additional text and changes in further sections.

File: contributors/devel/adding-an-APIGroup.md (hunk 2 above)
Comment (lavalamp) [inline contributors/devel/adding-an-APIGroup.md]: It turned into the registered package.

File: contributors/devel/adding-an-APIGroup.md [hunk 3]
```diff
@@ -21,21 +21,38 @@ in your group;
...
 
+### Type definitions in `types.go`
+
+Each type should be an exported struct (have a capitalized name). The struct
+should have the `TypeMeta` and `ObjectMeta` embeds. There should be a `Spec` and
+a `Status` field. If the object is soley a data storage object, and will not be
+modified by a controller, the status field can be left off and the fields inside
+the `Spec` can be inlined directly into the struct.
+
+For each type there should also be a `List` struct. The `List` struct should
```
Comment (lavalamp) [inline contributors/devel/adding-an-APIGroup.md]: Nit: I would qualify with "top-level type".

**Comment by lavalamp** [comment]
lgtm if you fix that nit.

File: contributors/devel/adding-an-APIGroup.md (hunk 3 above)
Comment (MHBauer) [inline contributors/devel/adding-an-APIGroup.md]: @lavalamp agree. done.

**Comment by lavalamp** [comment]
Thanks a bunch!

**Comment by MHBauer** [comment]
@lavalamp Thanks for feedback. I'll be around soon with an apiserver specific change.

**Pull request merged**